### ISSUE TITLE
Running rke remove --dind does not remove cluster.rkestate

### ISSUE THREAD

**Issue by dnoland1** [rancher/rke#1291]
**RKE version:**
```
vagrant@rke-node1:~$ rke -v
rke version v0.2.1
```
**Docker version: (`docker version`,`docker info` preferred)**
```
vagrant@rke-node1:~$ docker info
Containers: 13
 Running: 4
 Paused: 0
 Stopped: 9
Images: 8
Server Version: 18.09.2
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9754871865f7fe2f4e74d43e2fc7ccd237edcbce
runc version: 09c8266bf2fcf9519a651b04ae54c967b9ab86ec
init version: v0.18.0 (expected: fec3683b971d9c3ef73f284f176672c44b448662)
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-47-generic
Operating System: Ubuntu 18.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 985.5MiB
Name: rke-node1
ID: 6ADW:R3CP:PQF6:UAEE:IXQG:JTEV:RF5F:SRZ3:LWE6:4H5L:TBXD:2C3M
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
vagrant@rke-node1:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.2 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VirtualBox

**cluster.yml file:**
```
vagrant@rke-node1:~$ cat cluster.yml
nodes:
- address: node1
  user: docker
  role:
  - controlplane
  - etcd
  - worker
```
**Steps to Reproduce:**
1. run `rke up -dind`
2. run `rke remove -dind -force`

**Results:**
```
vagrant@rke-node1:~$ rke up --dind
INFO[0000] [dind] Pulling image [docker:17.03-dind] on host [unix:///var/run/docker.sock]
...
INFO[0195] Finished building Kubernetes cluster successfully
vagrant@rke-node1:~$ ls -l
total 112
-rw-r----- 1 vagrant vagrant 99128 Apr 15 16:31 cluster.rkestate
-rw-r--r-- 1 vagrant vagrant    84 Apr 15 16:28 cluster.yml
-rw-r----- 1 vagrant vagrant  5381 Apr 15 16:29 kube_config_cluster.yml
vagrant@rke-node1:~$ rke remove --dind --force
INFO[0000] [dind] Removing dind container [rke-dind-node1] on host [unix:///var/run/docker.sock]
INFO[0006] [dind] Successfully Removed dind container [rke-dind-node1] on host [unix:///var/run/docker.sock]
INFO[0006] Removing local admin Kubeconfig: ./kube_config_cluster.yml
INFO[0006] Local admin Kubeconfig removed successfully
vagrant@rke-node1:~$ ls -l
total 104
-rw-r----- 1 vagrant vagrant 99128 Apr 15 16:31 cluster.rkestate
-rw-r--r-- 1 vagrant vagrant    84 Apr 15 16:28 cluster.yml
```
Failing to remove the `cluster.rkestate` file appears to cause serious problems if you modify cluster.yml and re-run rke up. As a workaround, the cluster.rkestate file can be manually removed.

**Comment by jiaqiluo** [comment]
the bug is reproduced in rke v0.2.1

Steps:
- get an ec2 Linux instance
- create the cluster.yml file in the host
```
nodes:
- address: private-ip 
  user: docker
  role:
  - controlplane
  - etcd
  - worker
```
- run `rke up -dind`
- run `rke remove -dind -force`

result:
the file `cluster.rkestate` is left

**Comment by jiaqiluo** [comment]
the bug fix is validated in RKE v1.2.0-rc14

the steps are the same, and only one extra step is 
```
export DOCKER_API_VERSION=1.24
```
result:
the file `cluster.rkestate` is removed
